pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -28,75 +28,59 @@ import (
+func hotplugVirtioInterface(vmi *v1.VirtualMachineInstance, dom cli.VirDomain, domainInterfaces []api.Interface, domain *api.Domain) error {
 	vmConfigurator := netsetup.NewVMNetworkConfigurator(vmi, cache.CacheCreator{})
-	for _, network := range networksToHotplugWhoseInterfacesAreNotInTheDomain(vmi, indexedDomainInterfaces(currentDomain)) {
-		log.Log.Infof("will hot plug %s", network.Name)
+	for _, networkName := range networksToHotplug(vmi.Status.Interfaces, domainInterfaces) {
+		log.Log.Infof("will hot plug %s", networkName)
 
-		if err := vmConfigurator.SetupPodNetworkPhase2(updatedDomain, []v1.Network{network}); err != nil {
+		network := netvmispec.LookupNetworkByName(vmi.Spec.Networks, networkName)
+		if err := vmConfigurator.SetupPodNetworkPhase2(domain, []v1.Network{*network}); err != nil {
 			return err
 		}
 
-		relevantIface := domainInterfaceFromNetwork(updatedDomain, network)
+		relevantIface := domainspec.LookupInterfaceName(domainInterfaces, networkName)
 		if relevantIface == nil {
-			return fmt.Errorf("could not retrieve the api.Interface object from the dummy domain")
+			return fmt.Errorf("could not retrieve %q Interface object from domain", networkName)
 		}
 
-		log.Log.Infof("will hot plug %s with MAC %s", network.Name, *relevantIface.MAC)
+		log.Log.Infof("will hot plug %s with MAC %s", networkName, relevantIface.MAC)
 		ifaceXML, err := xml.Marshal(relevantIface)
 		if err != nil {
 			return err
 		}
 
 		if err := dom.AttachDeviceFlags(strings.ToLower(string(ifaceXML)), affectLiveAndConfigLibvirtFlags); err != nil {
-			log.Log.Reason(err).Errorf("libvirt failed to attach interface %s: %v", network.Name, err)
+			log.Log.Reason(err).Errorf("libvirt failed to attach interface %s: %v", networkName, err)
 			return err
 		}
 	}
 	return nil
 }
 
-func domainInterfaceFromNetwork(domain *api.Domain, network v1.Network) *api.Interface {
-	for _, iface := range domain.Spec.Devices.Interfaces {
-		if iface.Alias.GetName() == network.Name {
-			return &iface
-		}
-	}
-	return nil
-}
+// networksToHotplug returns the network names that available in the virt-launcher pod but not attached to the domain
+func networksToHotplug(vmiStatusIfaces []v1.VirtualMachineInstanceNetworkInterface, domainInterfaces []api.Interface) []string {
+	var networkNames []string
 
-func networksToHotplugWhoseInterfacesAreNotInTheDomain(vmi *v1.VirtualMachineInstance, indexedDomainIfaces map[string]api.Interface) []v1.Network {
-	var networksToHotplug []v1.Network
-	interfacesToHoplug := netvmispec.IndexInterfacesFromStatus(
-		vmi.Status.Interfaces,
-		func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool {
-			_, exists := indexedDomainIfaces[ifaceStatus.Name]
+	lookupDomainIfaceWithName := map[string]api.Interface{}
+	for _, domainIface := range domainInterfaces {
+		lookupDomainIfaceWithName[domainIface.Alias.GetName()] = domainIface
+	}
```

Comment:
I don't think I understood the value proposal of your suggestions. 

In https://github.com/kubevirt/kubevirt/pull/9399#discussion_r1150421006 you mention this would save some indexing. I still see indexing here.

Furthermore, if you're indexing, why do it inline ? What's wrong with encapsulating that into a function ?

Reply:
> In https://github.com/kubevirt/kubevirt/pull/9399#discussion_r1150421006 you mention this would save some indexing. I still see indexing here.
 
It turns out that it's not possible to avoid indexing.

> Furthermore, if you're indexing, why do it inline ? What's wrong with encapsulating that into a function ?

The indexing function used only at caller and did not used in other places, also I find it easier to follow the critical logic when its right in front of you.
But I am not insisting on it  I can bring it back.